### PR TITLE
Use static storage account for Form Recognizer live tests 

### PR DESCRIPTION
**Pull request by samvaity**
- Update tests to use static storage files.
- Remove storage package test dependency
- Add changelog updates.
- Fix flakey live tests 
Live test pipeline - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=379736&view=logs&j=4d5db6ce-0b7f-527e-b115-2367ee6e1fef&t=4d5db6ce-0b7f-527e-b115-2367ee6e1fef
- Remove static sample test and training files from repo

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java [hunk 1]
```diff
@@ -306,15 +277,18 @@ String getApiKey() {
...
+        return interceptorManager.isPlaybackMode()
+            ? "https://isPlaybackmode"
+            : Configuration.getGlobalConfiguration().get(FORM_RECOGNIZER_TRAINING_BLOB_CONTAINER_SAS_URL);
+    }
 
-        }
+    private String getTestingSasUri() {
+        return interceptorManager.isPlaybackMode()
+            ? "https://isPlaybackmode"
+            : Configuration.getGlobalConfiguration().get(FORM_RECOGNIZER_TESTING_BLOB_CONTAINER_SAS_URL);
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java:292]: These two methods can simplify to be one generic method.
```
private String getSasUri(String sasUrl) {}

``

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java [hunk 2]
```diff
@@ -124,72 +95,72 @@ private static void validateErrorData(List<ErrorInformation> expectedErrors,
...
+     * Deserialize test data from service.
+     *
+     * @return the deserialized raw response test data
+     */
+    static <T> T deserializeRawResponse(SerializerAdapter serializerAdapter, NetworkCallRecord record, Class<T> clazz) {
+        try {
+            return serializerAdapter.deserialize(record.getResponse().get("Body"),
+                clazz, SerializerEncoding.JSON);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to deserialize service response.");
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java:114]: do you want to user logger? or is that fine to throw it directly in test?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java (hunk 1 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java:292]: they are returning different configurations

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java (hunk 2 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java:114]: Logger in tests?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java [hunk 3]
```diff
@@ -527,14 +557,63 @@ String getApiKey() {
     protected String getEndpoint() {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java:557]: I saw this method twice in this PR.  Do we want to put it in TestUtil?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java:557]: Cannot move them outside of the TestBase classes as they the `interceptorManager` class instance to find the corresponding configurations.

**Pull request merged**